Test/Unit/Helper/DataTest.php

```diff
@@ -0,0 +1,141 @@
+            ->getMockForAbstractClass();
+        
+        $this->data = new Data($this->config, $this->storeManager);
+    }
+    
+    public function testGetStoreId()
+    {
+        $this->storeManager->expects($this->any())->method('getStore')->will($this->returnSelf());
+        $this->storeManager->expects($this->any())->method('getId')->will($this->returnValue('1'));
+        
+        $this->assertEquals(1, $this->data->getStoreId());
+    }
+    
+    public function testIsEnabled()
+    {
+        $this->config->expects($this->any())->method('getValue')
+            ->with(
+                $this->equalTo('metrilo_analytics/general/enable'),
+                $this->equalTo('store'),
+                $this->equalTo($this->storeId))
+            ->will($this->returnValue(true));
+        
+        $this->assertTrue($this->data->isEnabled($this->storeId));
+    }
+    
+    public function testGetApiToken()
+    {
+        $this->config->expects($this->any())->method('getValue')
+            ->with(
+                $this->equalTo('metrilo_analytics/general/api_key'),
+                $this->equalTo('store'),
+                $this->equalTo($this->storeId))
+            ->will($this->returnValue('9b4dd74a736d9d7d'));
+    
+        $this->assertEquals('9b4dd74a736d9d7d', $this->data->getApiToken($this->storeId));
+    }
+    
+    public function testGetApiSecret()
+    {
+        $this->config->expects($this->any())->method('getValue')
+            ->with(
+                $this->equalTo('metrilo_analytics/general/api_secret'),
+                $this->equalTo('store'),
+                $this->equalTo($this->storeId))
+            ->will($this->returnValue('82535e6593b51afed58e0a5a'));
+    
+        $this->assertEquals('82535e6593b51afed58e0a5a', $this->data->getApiSecret($this->storeId));
+    }
+    
+    public function testGetApiEndpoint()
+    {
+        $this->config->expects($this->any())->method('getValue')
+            ->with($this->equalTo('metrilo_analytics/general/api_endpoint'))
+            ->will($this->returnValue('https://trk.mtrl.me'));
+    
+        $this->assertEquals('https://trk.mtrl.me', $this->data->getApiEndpoint());
+    }
+    
+    public function testGetActivityEndpoint()
+    {
+        $this->config->expects($this->any())->method('getValue')
+            ->with($this->equalTo('metrilo_analytics/general/activity_endpoint'))
+            ->will($this->returnValue('https://p.metrilo.com'));
+    
+        $this->assertEquals('https://p.metrilo.com', $this->data->getActivityEndpoint());
+    }
+    
+    public function testLogError()
+    {
+        $logLocation = BP . '/var/log/metrilo.log';
+        if (file_exists($logLocation)) {
+            unlink($logLocation);
+        }
+        
+        $this->data->logError('test');
+        $result = file_exists($logLocation);
+        
+        $this->assertTrue($result);
+        $this->assertFalse(filesize($logLocation) > 10 * 1024 * 1024);
```

Comment:
You will never come to a moment that this will be `true`. Maybe have a test that creates a file big enough?

Reply:
I was unable to come up with a solution how to properly test this. Had that idea to include a file with wall of text big enough to fill 10GB of disc space but in the end tough it's unnecessary.

Reply:
wait, wait... `10 * 1024 * 1024` is 10Mb, right? We don't want to have 10Gb of log file on client's server.

Reply:
Ohh yes, i mean 10 MB not GB. Sorry for the typo.

Reply:
Yeah, BIG typo.